[Zhehao] iP

Charles1026 opened the pull request:
## Naruto ChatBot

> “HARD WORK IS WORTHLESS FOR THOSE THAT DON’T BELIEVE IN THEMSELVES.” – NARUTO UZUMAKI

Naruto frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,
- download it from [here](https://github.com/Charles1026/ip).
- double-click it.
- add your tasks.
- let it manage your tasks for you 😉

And it is FREE!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Duke.class, args);
    }
}
```

Commit:
Add Gradle support

Charles1026 committed:
Level 1

Charles1026 committed:
Level 2

Charles1026 committed:
Level 3

Charles1026 committed:
Level 4

Charles1026 committed:
A-TextUiTesting

Charles1026 committed:
Level 5

Charles1026 committed:
Level 6

Charles1026 committed:
Level 7

bensohh commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Item{
```
Maybe it would look cleaner to add a spacing between Item and the '{'?

bensohh commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Item{
+    private final String itemType = "[D]";
```
Since itemType is a final variable similar to a constant, would it be better to use the constant naming convention?

bensohh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,112 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        System.out.println("Hello! I'm Naruto and one day I will be Hokage! \nWhat can i do for you?");
```
I really like the unique concept of Naruto here!

bensohh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,112 @@
+            String rawInput = scanner.nextLine().trim();
+            String[] input = rawInput.split(" ", 2);
+            System.out.println(lineBreak);
+            switch (input[0]) {
+                case "list":
```
Perhaps it will look neater to follow the switch-case indentation (same-level) from the coding standards?

bensohh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,112 @@
+                    break;
+
+                case "help":
+                    System.out.println("    Hi are you in need of help? These are the commands you can use Dattebayo:" +
+                            "\n      todo <Task>: add a task you need to do" +
+                            "\n      deadline <Task> /by<Deadline>: add a deadline you need to complete" +
+                            "\n      event <Task> /at<Time>: add an event you need to attend" +
+                            "\n      mark <index>: mark an task as completed by index" +
+                            "\n      unmark <index>: unmark a completed task by index" +
+                            "\n      list: view all you tasks by index" +
```
Maybe it would be neater to do the line break and place the "+" symbol at the start of the next line instead of the end of previous line. Similar cases spotted above.

bensohh commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,14 @@
+public class Event extends Item{
```
Similarly, would be good to add a spacing between Item and "{", there are a few similar cases spotted below.

bensohh commented on src/main/java/Item.java:
```diff
@@ -0,0 +1,31 @@
+
+    @Override
+    public String toString() {
+        return (this.isDone ? this.done : this.unDone) + this.item;
```
Very good use of ternary operator expressions that is similar to the coding standards!

bensohh commented on src/main/java/Naruto.java:
```diff
@@ -0,0 +1,79 @@
+        return "    Looks like this is the end for now, till we meet again! Ja Ne!";
+    }
+
+    public static int string2Int(String input) throws NumberFormatException {
```
Perhaps it might be better to change this to stringToInt?

JJoeYi replied:
Perhaps it might be better to change this to stringToInt?

bensohh reviewed (commented):
LGTM 👍  Just require some minor edits! Good job, I really like the modification to Duke 😄

JJoeYi commented on src/main/java/Naruto.java:
```diff
@@ -0,0 +1,79 @@
+        this.storedItems = new ArrayList<>(100);
+    }
+
+    private String addItem(Item item) {
```
I like the way you named your methods, by using verbs for actions I can clearly understand what the methods do without having to read the implementation.

JJoeYi commented on src/main/java/Item.java:
```diff
@@ -0,0 +1,31 @@
+public abstract class Item {
+    private boolean isDone = false;
+    private final String done = "[X] ";
```
If these strings are final and constants, perhaps change the naming to be all CAPS?

JJoeYi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,112 @@
+        Naruto naruto = new Naruto();
+        String lineBreak = "-".repeat(20);
+
+        boolean breakLoop = false;
```
Perhaps considering changing the name of this boolean to something that sounds more like a check? e.g. isBreakLoop

JJoeYi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,112 @@
+
+                case "deadline":
+                    try {
+                        String[] splitInput = input[1].split("/by", 2);
```
I like the way you named this variable. But perhaps consider changing it to splitInputBy for "deadline" case and splitInputAt for "event' case?

JJoeYi reviewed (commented):
Some nits to fix.

Charles1026 committed:
Level 7

Charles1026 committed:
Level 8

Charles1026 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

Charles1026 committed:
A-Gradle

Charles1026 committed:
A-MoreOOP

Charles1026 committed:
A-JUnit

Charles1026 committed:
A-Packages

Charles1026 committed:
A-Jar

Charles1026 committed:
A-Packages

Charles1026 committed:
Revert "A-Packages"

This reverts commit c0b2e5cff6dc1087f47a2c0c929a928463f98bf6.

Charles1026 committed:
Merge branch 'master' into HEAD

Charles1026 committed:
A-JavaDoc

Charles1026 committed:
A-CodingStandard

Charles1026 committed:
Level-9

Charles1026 committed:
A-CheckStyle

Charles1026 committed:
Level-10

Charles1026 committed:
A-Assertions: add assert method and file checker to storage.

The loadItems method has no check to ensure the tasks file is present.

The loadItems method cannot function without a task file present.

A function to create the file if it is not there and an assert method
to ensure the file is there will be added.

A function and assert are both used as assert cannot be used to carry
out tasks and should be used just as a check

Charles1026 committed:
A-CodeQuality: Split Duke start into multiple functions

Duke start function is very long currently

The code is quite unreadable and can be easily abstracted

Let's abstract the code into individual java fx components

Make the code readable and it follows logical sense

Charles1026 committed:
A-CodeQuality: Split Duke start into multiple functions

Duke start function is very long currently

The code is quite unreadable and can be easily abstracted

Let's abstract the code into individual java fx components

Make the code readable and it follows logical sense

Charles1026 committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/duke/Duke.java

Charles1026 committed:
C-DetectDuplicates

Charles1026 committed:
C-DetectDuplicates

Charles1026 committed:
Merge branch 'branch-C-DetectDuplicates'

Charles1026 committed:
Bug Fix and Ui Addition

Charles1026 committed:
PR

Charles1026 committed:
Merge pull request #2 from Charles1026/branch-PR

PR

Charles1026 committed:
PR

Charles1026 committed:
Merge pull request #3 from Charles1026/A-UserGuide

User guide

Charles1026 committed:
Actual User Guide

Charles1026 committed:
Merge pull request #4 from Charles1026/branch-A-UserGuide

Actual User Guide

Charles1026 committed:
Credit Library and build.gradle changes
